### DISCUSSION TITLE
Script - How to use a for loop to iterate over telemetry data

### DISCUSSION THREAD

**Discussion by letshin** [arendst/Tasmota#21027] [Q&A] [answered]
HI. I'm trying to shorten my script below using a `for` loop:
```
>T
t1=DS18B20#Temperature
if t1!=0 {
        t+=t1
        cnt+=1
}
t1=DS18B20_1#Temperature
if t1!=0 {
        t+=t1
        cnt+=1
}
t1=DS18B20_2#Temperature
if t1!=0 {
        t+=t1
        cnt+=1
}
t1=DS18B20_3#Temperature
if t1!=0 {
        t+=t1
        cnt+=1
}
t1=DS18B20_4#Temperature
if t1==0 {
        t+=t1
        cnt+=1
}
tOut=t/cnt
t=0
cnt=0
```
I've tried:
```
>T
for cnt 1 4 1
  temp1=s(1.0cnt) 
  print %temp1%
  temp2=DS18B20_%temp1%#Temperature
  print %temp2%
next
```
I've defined the variables in `>B` correctly but have left them out here for the sake of brevity. I've used `s(1.0cnt)` since otherwise `print %cnt%` returns 1.0, 2.0, 3.0 and so on. Although the above returns `temp1=1...4` and hence DS18B20_1#Temperature and so on and so forth, it doesn't give me the DS18B20#Temperature results. Is this possible to do with a for loop at all?

**Comment by gemu2015** [comment, marked as answer]
sorry, substituting variables in a json path is currently not supported

**Reply by letshin** [reply]
Thanks. Would there be any other way to shorten the code or is what I've done pretty much it?

**Reply by gemu2015** [reply]
you could read each value into an array and then evaluate the array in a loop
```
>D
M:array=0 4
cnt=0
temp1=0

>T

; use array in auto increment mode
array[0]=0
array=DS18B20#Temperature
array=DS18B20_2#Temperature
array=DS18B20_3#Temperature
array=DS18B20_4#Temperature
for cnt 1 4 1
  temp1=array[cnt]
  print %temp1%
next

```
